stats/bridge_stats.go

```diff
@@ -258,41 +274,20 @@ func (s *Service) report(conn *connWrapper) error {
 	report := map[string][]interface{}{
 		"emit": {"bridge-stats", info},
 	}
+	log.Info("[Bridge-Stats] Emit Bridge stats")
 	return conn.WriteJSON(report)
 }
 
-func (s *Service) reportLoop(conn *connWrapper) {
-	sendStatsTicker := time.NewTicker(10 * time.Second)
-	for {
-		select {
-		case <-s.quitCh:
-			return
-		case msg := <-s.errCh:
-			s.setLastError(msg.Listener, msg.Err)
-		case msg := <-s.processedBlockCh:
-			s.setProcessedBlock(msg.Listener, msg.ProcessedBlock)
-		case <-sendStatsTicker.C:
-			if err := s.report(conn); err != nil {
-				log.Warn("bridge stats report failed", "err", err)
-				return
-			}
-		}
-	}
-}
-
 // readLoop loops as long as the connection is alive and retrieves data packets
 // from the network socket. If any of them match an active request, it forwards
 // it, if they themselves are requests it initiates a reply, and lastly it drops
 // unknown packets.
 func (s *Service) readLoop(conn *connWrapper) {
 	// If the read loop exits, close the connection
 	defer conn.Close()
+	log.Info("[Bridge stats] Start read loop")
 	for {
-		select {
-		case <-s.quitCh:
```

Comment:
this select is used to stop when quitCh is closed, otherwise the loop will keep running.
idk why you remove these lines

Reply:
```
// Start loop keeps trying to connect to ronin stats server, reporting bridge stats.
func (s *Service) Start() {
	errTimer := time.NewTimer(0)
	defer errTimer.Stop()
	// Loop reporting until termination
	for {
		select {
		case <-s.quitCh:
			return
```
I see we handle the signal quit.Ch in start function ( the function start go routine readLoop), hm In case the the start exist, the go routine inside it should exist too or not ? let me check this case.

Reply:
checked, then you are right